src/components/Preferences/LocalIdentities.js

```diff
@@ -19,132 +20,244 @@ import {
 } from '../IdentityManager/IdentityManager'
 import EmptyLocalIdentities from './EmptyLocalIdentities'
 import Import from './Import'
+import { GU } from '../../utils'
 
 const iOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream
 
-const LocalIdentities = ({
-  dao,
-  localIdentities,
-  onClearAll,
-  onImport,
-  onModify,
-  onModifyEvent,
-}) => {
-  // transform localIdentities from object into array and attach address to each entry
-  const identities = Object.entries(localIdentities).map(
-    ([address, identity]) => ({
-      ...identity,
-      address,
-    })
-  )
+const SelectableLocalIdentities = React.memo(
+  ({ localIdentities, ...props }) => {
+    const identities = React.useMemo(
+      () =>
+        Object.entries(localIdentities).map(([address, identity]) => ({
+          ...identity,
+          address,
+        })),
+      [localIdentities]
+    )
+    const initialAddressesSelected = React.useMemo(
+      () => new Map(identities.map(({ address }) => [address, true])),
+      [identities]
+    )
+    const [addressesSelected, setAddressesSelected] = React.useState(
+      initialAddressesSelected
+    )
+
+    const handleToggleAll = React.useCallback(
+      () =>
+        setAddressesSelected(
+          new Map(
+            identities.map(({ address }) => [
+              address,
+              !(
+                Array.from(addressesSelected.values()).every(v => v) ||
+                Array.from(addressesSelected.values()).some(v => v)
```

Comment:
I noticed we were doing this below as well, what do you think about moving the `useMemo()` declarations up to here and then we can pass it down through props?

Reply:
I know, but I ended up not adding this as props because it just felt like clutter, I mean adding two extra bool props to tell the Component if all or some of the addresses have been selected? When it is something that can be computed and memoized?
I may be convinced but you'll have to convince me ;)

Reply:
I guess the _most_ react way of doing this would be to abstract the logic into a separate memoized function or hook, but it seems a bit extreme for this case so let's leave it 😄